### ISSUE TITLE
Player wears and removes hood/collar/pockets constantly under certain conditions

### ISSUE THREAD

**Issue by Shoes01** [CleverRaven/Cataclysm-DDA#10966]
I think this is triggered by wearing a single super warm item. I started a game with a winter coat and the UI flips between "(falling)" and "(rising)", probably because the player gets too warm when a body part (hands, mouth, head) gets too warm due to the bonus pockets/collar/hood. I suppose to fix this, the code would have to want warm hands over cold ones, meaning it would have to guess at what the temp would be if it removed the hands. 

Take this with a grain of salt, I am working on unrelated stuff but noticed this. I also got a segfault on spawning a thermometer, so maybe my build is messed! :D

**Comment by kevingranade** [comment]
This is a known issue, the problem is you put up the hood, warm up a tiny
bit the next turn, take it down, cool off the next turn, repeat.  One
solution for this would be to have the point where you put it up and take
it down be different, then it would at least take a while between changes.

**Comment by Shoes01** [comment]
I figured it was known; should have searched for it first.

I think if we make the window of acceptable temperatures greater, we might alleviate this problem. At the moment, the code reads :

> if ( ~ && ( temp_conv[bp] <= BODYTEMP_NORM || temp_cur[bp] <= BODYTEMP_COLD) ) 

which says "If you will be close to cold or you are right now cold" you will wear hoods/etc. 

I suggest we change it to : 

> if ( ~ && ( temp_conv[bp] <= BODYTEMP_WARM || temp_cur[bp] <= BODYTEMP_COLD) ) 

This says the player will put on the hood/etc even if he will be in the high end of the comfort range. This does not fix the issue of predicting player temperature, but it makes it less likely to happen, because you might not be wearing a winter coat if you don't have to. 

As for actually trying to fix this problem, I would think about moving the hoods/etc code outside of warmth, and calculating it after the bodytemp has been calculated for that turn. So the last line in player::update_bodytemp() could read "if (temp_conv[i] < BODYTEMP_COLD) {use_pockets_etc(bp)}".

**Comment by kevingranade** [comment]
Widening the existing range isn't going to fix it, the idea is to replace
it with two checks that trigger at different values.
```
If( hood_up && ( temp_conv[bp] >= BODYTEMP_NORM + 10 || ... ) {
    put_hood_down():
} else if( hood_down && ( temp_conv[bp] <= BODYTEMP_NORM - 10 || ... ) {
    put_hood_up();
}
```
Then you have the time period between the two as a natural delay.  You
might even use the boundaries of comfort if you want the range to be really
wide.

**Comment by Shoes01** [comment]
Well, what I am looking at now might work. I've moved the bonus warmth provided by hoods/etc outside of the warmth function, so now I can look at their impact on the player's warmth. After that, I just have to check "If this warmth is added, will it converge to too high a value? OR am I cold right now?". 

I can't push anything from work, but here is the pastebin from the piece I've added : http://pastebin.com/KaqtMB9Y

The key is "clothing_warmth_adjusted_bonus", which calls a new function that looks at only the warmth provided by hoods/etc. 

I get excited when I fix things :3
